server/src/framework.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  ConnInfo,
+  Handler,
+} from "https://deno.land/std@0.182.0/http/server.ts";
+
+type MethodHandler = { POST?: Handler; GET?: Handler };
+
+interface Router {
+  [index: string]: MethodHandler;
+}
+
+export const app = (router: Router): Handler => {
+  const routes: Array<[RegExp, MethodHandler]> = Object.entries(router).map(
+    ([url_pattern, handler]) => [new RegExp(`^${url_pattern}$`), handler]
+  );
+
+  return (
+    request: Request,
+    connInfo: ConnInfo
+  ): Response | Promise<Response> => {
+    for (const route of routes) {
+      const [pattern, methodHandler] = route;
+      const pathname = new URL(request.url).pathname;
+      const myMatch = pathname.match(pattern);
+      if (myMatch == undefined) {
+        continue;
+      }
+      const requestHandler =
+        methodHandler[request.method as keyof MethodHandler];
+      if (requestHandler == undefined) {
+        continue;
+      }
+      return requestHandler(request, connInfo);
+    }
+    return Response404();
+  };
+};
+
+export const Response401 = () =>
```

Comment:
i couldn't spot where this is used except this file

Reply:
It's not, but I think it reads better this way. We also have a Response404 which is used outside. Should I remove this?

Reply:
i think so, i usually use export to know what i can refactor/change/rename without looking at other files